src/Services/InteractsWithPage.php

```diff
@@ -0,0 +1,197 @@
+    }
+
+    /**
+     * Assert the page is at the path that you specified.
+     *
+     * @param $path
+     *
+     * @return $this
+     */
+    protected function seePageIs($path)
+    {
+        $this->assertEquals($this->baseUrl.$path, $this->url());
+
+        return $this;
+    }
+
+    /**
+     * Type a value into a form input by that inputs name.
+     *
+     * @param $value
+     * @param $name
+     * @param bool $clear Whether or not to clear the input first on say an edit form
+     *
+     * @return $this
+     */
+    protected function type($value, $name, $clear = false)
+    {
+        $element = $this->findElement($name);
+
+        if ($clear) {
+            $element->clear();
+        }
+
+        $element->value($value);
+
+        return $this;
+    }
+
+    /**
+     * Function to type information as an array
+     * The key of the array specifies the input name.
+     *
+     * @param $information
+     * @param $clear
+     *
+     * @return $this
+     */
+    protected function type_information($information, $clear = false)
+    {
+        foreach ($information as $element => $item) {
+            $this->type($item, $element, $clear);
+        }
+
+        return $this;
+    }
+
+    protected function submitForm($inputs, $selector)
+    {
+        $form = $this->byCssSelector($selector);
+        $this->type_information($inputs);
+        $form->submit();
+
+        return $this;
+    }
+
+    /**
+     * Press a button on the page that contains text.
+     *
+     * @param $text
+     *
+     * @return $this
+     */
+    protected function press($text)
+    {
+        $this->findElement($text, "//button[contains(text(), '{$text}')]")->click();
+
+        return $this;
+    }
+
+    /**
+     * Click an element based on text passed in, or pass an Id or Name to find the element by.
+     *
+     * @param $textOrId
+     *
+     * @throws CannotClickElement Throws when the element cannot be clicked
+     *
+     * @return $this
+     */
+    protected function click($textOrId)
+    {
+        $element = $this->findElement($textOrId, "//a[contains(text(), '{$textOrId}')]");
+
+        try {
+            $element->click();
+        } catch (\Exception $e) {
+            throw new CannotClickElement('Cannot click the element with the text: '.$textOrId);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Will attempt to find an element by different patterns
+     * If xpath is provided, will attempt to find by that first.
+     *
+     * @param $value
+     * @param null $xpath
+     *
+     * @throws CannotFindElement
+     *
+     * @return \PHPUnit_Extensions_Selenium2TestCase_Element
+     */
+    protected function findElement($value, $xpath = null)
+    {
+        $element = null;
+        try {
+            if (!is_null($xpath)) {
+                return $this->byXPath($xpath);
+            }
+        } catch (\Exception $e) {
+        }
```

Comment:
I don't catch the exception here simply because ```findElement``` is a function that uses multiple methods of getting an element, based on what you pass in.  If the element can't be found in the end, an exception is thrown anyway.  

This simplifies the API into getting elements for the page interaction functions, even those may use an XPath selector like ```"//a[contains(text(), '{$textOrId}')]"```

I agree that this doesn't allow for specific finding of elements though.  That can still be done by the end user in their test 

->byName($value), ->byId($value), ->byCssSelector($value), ->byXPath($xpath)